packages/ERTP/src/purse.js

```diff
@@ -1,43 +1,60 @@
 import { makeNotifierKit } from '@agoric/notifier';
-import { Far } from '@endo/marshal';
+import { defineKind } from '@agoric/swingset-vat/src/storeModule.js';
 import { AmountMath } from './amountMath.js';
 
-export const makePurse = (allegedName, assetKind, brand, purseMethods) => {
-  let currentBalance = AmountMath.makeEmpty(brand, assetKind);
+export const makePurseMaker = (allegedName, assetKind, brand, purseMethods) => {
+  const makePurseKit = defineKind(
+    allegedName,
```

Comment:
Shouldn't the allegedName include a description of the kind?

What is the kind being defined here? From `makePurseKit` I read this as a "purse kit" kind, but I'm surprised to see a kit be a durable thing with state. I thought we used kit as a collection of related functionality.

It looks like this was made to have a place outside `purse` to the deposit facet but I don't see the motivation for that. Especially since `purse` still has `getDepositFacet`.

Reply:
> Shouldn't the allegedName include a description of the kind?

The "kind" is the pair of the `purse` and `depositFacet`, of which only the `purse` is immediately exposed.

Since we're calling (the maker returned by) `defineKind` with (a function that returns) a record with both `purse` and `depositFacets`, and only these objects are declared `Far`, `defineKind` with call `Far` with farNames of, respectively 
```js
`${allegedName} purse`
```
and
```js
`${allegedName} depositFacet`
```
which, somewhat fortuitously, actually mimics its previous behavior.

Reply:
> It looks like this was made to have a place outside purse to the deposit facet but I don't see the motivation for that. Especially since purse still has getDepositFacet.

The `purse` and `depositFacet` need each other, but both are exposed to the outside world as independent `Far` objects. However, between the two of them, only one state record is needed. This solution doesn't pay any storage in the `state` record for the `depositFacet` because it only needs to exist at runtime, reactivated at the same time the purse is. But it still preserves the identity of the per-purse depositFacet as a Far object.

Do you have another solution in mind?

Reply:
> Do you have another solution in mind?

Not yet; trying to understand this:

> both are exposed to the outside world as independent Far objects

In `master`, `depositFacet` is exposed only via `purse.getDepositFacet()`. Why the change to expose it as an independent Far object?  https://github.com/Agoric/agoric-sdk/blob/45723b48823ed59c647d4b1f4b4d7ae16c421209/packages/ERTP/src/purse.js#L35-L43

Reply:
The `depositFacet` is exposed *only* by `purse.getDepositFacet()`. But since it is exposed, it must be a far object. Since it has the same cardinality as purses, it must also be virtualized. But since it has no state independent of its purse, it need not be separately virtualized. It can just be virtualized as one of the cohort pair activated for reviving that state record.

Reply:
IIRC, we figured that `getDepositFacet()` *could* create a new facet on demand, which would just wrap the Purse and expose a deposit method, but 1: it made a good candidate for practicing the multi-facet objects, and 2: there may have been some maintain-a-constant-identity behavior that was easier/only-possible to achieve with the multi-facet approach.

Reply:
Ah. Since this code is meant to serve as an example I'd ask that some of the above be documented in it. If I've got this right,
```
export const makePurseMaker = (allegedName, assetKind, brand, purseMethods) => {
  // - This kind is a pair of purse and depositFacet that have a 1:1 correspondence.
  // - They are virtualized together to share a single state record.
  // - An alternative design considered was to have this return a Purse alone
  //   that created depositFacet as needed but this approach ensures a constant
  //   identity for the facet and exercises the multi-faceted object style.
  const makePurseKit = defineKind(
```

Reply:
Done, thanks.